pkg/cli/aws/aws.go

```diff
@@ -208,12 +239,88 @@ func refreshCredsViaOktaAWSCLI(ctx context.Context, copts *CredentialOptions, ac
 	return nil
 }
 
-// runCmd is a wrapper for running a command via exec.CommandContext
-// and passing through stdin/stdout/stderr.
-func runCmd(ctx context.Context, name string, args ...string) error {
-	cmd := exec.CommandContext(ctx, name, args...)
+// oktaAwsCLICmd is a convenience function that creates an exec.Cmd
+// instance for okta-aws-cli.
+func oktaAwsCLICmd(ctx context.Context, args ...string) *exec.Cmd {
+	cmd := exec.CommandContext(ctx, "okta-aws-cli", args...)
 	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
 	cmd.Stdin = os.Stdin
+	return cmd
+}
+
+// runOktaAwsCLI is a wrapper for running okta-aws-cli via exec.CommandContext,
+// passing through stdin/stdout/stderr, and setting the appropriate
+// environment variables.
+func runOktaAwsCLI(ctx context.Context, b *box.Config, args ...string) error {
+	cmd := oktaAwsCLICmd(ctx, args...)
+	cmd.Stdout = os.Stdout
+	// Always set up the Okta environment variables based off of what's
+	// in the box config, regardless of what's currently in the environment.
+	cmd.Env = cmd.Environ()
+	cmd.Env = append(
+		cmd.Env,
+		"OKTA_ORG_DOMAIN="+b.AWS.Okta.OrgDomain,
+		"OKTA_AWS_ACCOUNT_FEDERATION_APP_ID="+b.AWS.Okta.FederationAppID,
+		"OKTA_OIDC_CLIENT_ID="+b.AWS.Okta.OIDCClientID,
+	)
 	return cmd.Run()
 }
+
+// isOktaAwsCliVersion1 determines what major version of okta-aws-cli
+// is installed. If the CLI is not installed and in dry run mode,
+// return false (not v1).
+func isOktaAwsCliVersion1(ctx context.Context, cliExists bool) (bool, error) {
+	if !cliExists {
+		// Assumes that we're in dry run mode
+		return false, nil
+	}
+
+	cmd := oktaAwsCLICmd(ctx, "--version")
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return false, errors.Wrap(err, "cannot get stdout pipe")
+	}
+
+	if err := cmd.Start(); err != nil {
+		return false, errors.Wrap(err, "problem executing okta-aws-cli --version")
+	}
+
+	output, err := io.ReadAll(stdout)
+	if err != nil {
+		return false, errors.Wrap(err, "could not read version output")
+	}
+
+	if err := cmd.Wait(); err != nil {
+		return false, errors.Wrap(err, "problem completing execution of okta-aws-cli --version")
+	}
```

Comment:
Why not use [`cmd.Output`](https://pkg.go.dev/os/exec#Cmd.Output)?

Reply:
Because I didn't know it existed :sweat_smile: